shared/src/commonMain/kotlin/com/myworkout/shared/app/data/services/AuthenticationService.kt

```diff
@@ -0,0 +1,34 @@
+package com.myworkout.shared.app.data.services
+
+import com.myworkout.kotlin.api.apis.AuthApi
+import com.myworkout.shared.domain.core.Response
+
+interface AuthenticationService {
+
+    suspend fun authenticate(userName: String, password: String): Response<String>
+}
+
+class AuthenticationServiceImpl(
+    private val apiService: ApiService
+) : AuthenticationService {
+
+    private val authApi: AuthApi by lazy {
+        apiService.createAuthApi()
+    }
+
+    override suspend fun authenticate(userName: String, password: String): Response<String> {
+        return try {
+            val tokenResponse = authApi.login(userName, password)
+            val token = tokenResponse.body().data?.jwt
+
+            // very basic error handling here, but since it's a prototype ...
+            if (token.isNullOrEmpty()) {
+                Response.Failure(reason = Response.FailureReason.UNKNOWN)
```

Comment:
Would it be better to have an error category that's more descriptive/specific  here than a general `UNKNOWN `. I think all implementations in the whole return block only returns unknowns.

Reply:
I generally agree with you that it would be a good idea not just to return `UKNOWN` here, but as commented in line 24 for a prototype that's maybe good enough.

Reply:
Got it 👍